sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -1467,6 +1498,42 @@ void reduCGFunc(handler &CGH, KernelType KernelFunc,
   }
 }
 
+// Specialization for devices with the atomic64 aspect, which guarantees 64 (and
+// 32) bit floating point support for atomic add.
```

Comment:
This comment confuses me a little bit.  It sounds like if a device has `atomic64`, we'll select a different implementation of reductions for 32-bit floats.  Is that right?

Reply:
Sorry I should have explained this issue.  The default implementation for fp32 is currently using the deprecated atomic.fetch_add rather that atomic_ref.fetch_add.  Atomic.fetch_add doesn't support float according to the sycl2020 specification.  I guess that the operations in the reducer class will all be upgraded to using atomic_ref at some point once atomic_ref is fully implemented?  
I believe that currently atomic_ref is only supported for fetch_add.  The specification does not have any aspects/extensions that refer to atomic operations for fp32.  This seems to imply that, once atomic_ref is fully implemented in the reducer class operations, the boolean "has_fast_atomics" will be set true for the fp32 case for all inbuilt operations by default?  At that point the atomic64 specialization here will be uniquely for fp64, as its name suggests.  Currently I have also used it for the fp32 case.  
Alternatively I could immediately switch all usage of atomic.fetch_add to atomic_ref.fetch_add in the reducer class, so that atomics are used as default for the fp32 add operator reduction case, and then add a condition to "has_fast_atomics" for this case, i.e.:
```
(sycl::detail::is_sgenfloat<T>::value && 
 sycl::detail::IsPlus<T, BinaryOperation>::value &&
 sizeof(T) == 4)
```
This is perhaps outside the scope of this PR however?

Reply:
Thanks for the explanation, that all makes sense.

I agree that changing everything to use `atomic_ref` is the right long-term solution, and that it's outside the scope of this PR.  But it might be worth expanding this comment to say something like "32 bit floating point atomics are eventually expected to be supported by the has_fast_atomics specialization.", or to add a TODO somewhere.

Reply:
OK I'll add a comment, thanks.

Reply:
I have now made the change.